v2/server.go

```diff
@@ -185,14 +176,16 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Call the registered Intercept Function
-	if s.interceptFunc != nil {
-		req := s.interceptFunc(&RequestInfo{
-			Request: r,
-			Method:  method,
-		})
-		if req != nil {
-			r = req
+	// Call the registered Intercept Functions
+	if len(s.interceptFuncs) > 0 {
```

Comment:
You can remove all of the `len` checks in this PR. If a slice is empty, Go will just skip the loop.